File: rentPrediction/RentPrediction/Front End/ClientApp/src/app/models/properties/Property.ts
import { Address } from "../address/Address";
import { Feature } from "../feature/Feature";
import { Gallery } from "../galery/Gallery";
import { User } from "../users/User";
import { PropertyType } from "./PropertyType";

export class Property{
    id:number;
    name:string;
    description:string;
    url:string;
    price:number;
    predictedPrice:string;
    usableSurface:number;
    surface:number;
    creationDate:Date;
    availableDate:string;
    feature:Feature;
    address:Address;
    propertyType: PropertyType;
    user: User;
    userId:number;
    images:Gallery[];
}

File: rentPrediction/RentPrediction/RentPrediction.Data.Entities/Property.cs
using System;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;


namespace RentPrediction.Data.Entities
{
    public class Property : BaseEntity
    {
        public string Name { get; set; }
        public string Price { get; set; }
        public string PredictedPrice { get; set; }
        public string Description { get; set; }
        public int Surface { get; set; }
        public int UsableSurface { get; set; }
        public DateTime CreationDate { get; set; }
        public string URL { get; set; }
        public int AddressId { get; set; }
        public Address Address { get; set; }
        public int PropertyTypeId { get; set; }
        public PropertyType PropertyType { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public int FeatureId { get; set; }
        public Feature Feature { get; set; }
        public IList<Favorite> Favorites { get; set; }
        public IList<Gallery> Galleries { get; set; }
    }
    public class PropertyEntityConfiguration : IEntityTypeConfiguration<Property>
    {
        public void Configure(EntityTypeBuilder<Property> builder)
        {
            // Configure relationship with related entities
            builder.HasKey(x => x.Id);
            builder.HasMany(p => p.Favorites)
                .WithOne(f => f.Property)
                .HasForeignKey(f => f.PropertyId)
                .IsRequired()
                .OnDelete(DeleteBehavior.ClientCascade);
            builder.HasMany(p => p.Galleries)
                .WithOne(f => f.Property)
                .HasForeignKey(f => f.PropertyId)
                .IsRequired()
                .OnDelete(DeleteBehavior.ClientCascade);
       
            builder.ToTable("Properties");
        }
    }
}
